web3/middleware/stalecheck.py

```diff
@@ -0,0 +1,44 @@
+import time
+
+from web3.exceptions import StaleBlockchain
+
+SKIP_STALECHECK_FOR_METHODS = set([
+    'eth_getBlockByNumber',
+])
+
+
+def _isfresh(block, allowable_delay):
+    return block and time.time() - block['timestamp'] <= allowable_delay
+
+
+def make_stalecheck_middleware(allowable_delay=0):
```

Comment:
I think that this should not have a default value to ensure it's required that they specify a delay in their code.  (I get this is the current behavior but I think the validation should be separate).

Reply:
:man_facepalming: yeah, this was a remnant of the old arguments style.